scripts/globals/spells/enfeebling_song.lua

```diff
@@ -0,0 +1,423 @@
+    if resistancePotency < resistThreshold then
+        spell:setMsg(xi.msg.basic.MAGIC_RESIST)
+        return songEffect
+    end
+
+    ------------------------------
+    -- STEP 3: Calculate power, duration, tick, etc.
+    ------------------------------
+    local power    = xi.spells.enfeebling.calculateSongPower(caster, spellId, gearBoost, jobPointEffect, songEffect, soulVoiceType) or 0
+    local duration = xi.spells.enfeebling.calculateSongDuration(caster, spellId, gearBoost, jobPointEffect, songEffect, soulVoiceType) or 0
+
+    ------------------------------
+    -- STEP 4: Special cases.
+    ------------------------------
+    local finalValues = xi.spells.enfeebling.processSongExceptions(caster, target, spell, power, duration, dotTick, resistancePotency, songEffect)
+    if finalValues.earlyQuit then
+        return finalValues.effect
+    end
+
+    ------------------------------
+    -- STEP 5: Attempt to apply the status effect. Check for magic burst.
+    ------------------------------
+    if target:addStatusEffect(finalValues.effect, finalValues.power, finalValues.dotTick, finalValues.duration, 0, augmentStat) then
+        local _, skillchainCount = xi.magicburst.formMagicBurst(spellElement, target)
+        if skillchainCount > 0 then
+            spell:setMsg(xi.msg.basic.MAGIC_BURST_ENFEEB)
+            caster:triggerRoeEvent(xi.roeTrigger.MAGIC_BURST)
+        else
+            -- Lullaby has a different application message than the rest of the song debuffs.
+            if finalValues.effect == xi.effect.SLEEP_I then
+                spell:setMsg(xi.msg.basic.MAGIC_ENFEEB_IS)
+            else
+                spell:setMsg(xi.msg.basic.MAGIC_ENFEEB)
+            end
+        end
+    else
+        spell:setMsg(xi.msg.basic.MAGIC_NO_EFFECT)
+    end
+
+    return finalValues.effect
+end
+
+-----------------------------------
+-- Calculates song magic accuracy.
+-- TODO: Troubadour has an effect on magic accuracy with merits. Details unknown.
+-- Weather does not affect magic accuracy. (https://www.bg-wiki.com/ffxi/Weather)
+-----------------------------------
+---@param caster CBaseEntity
+---@param gearBoost integer the amount of Song+ and All Song from gear
+---@param spellId xi.magic.spell
+---@param spellElement xi.mod
+---@return number
+-----------------------------------
+xi.spells.enfeebling.calculateSongAccuracy = function(caster, gearBoost, spellId, spellElement)
+    local affectedByGear = pTable[spellId][11]
+    local totalAccuracy  = 0
+
+    -- Some spells types on BG Wiki don't explicitly say that Song+ gear affects magic accuracy.
+    if affectedByGear then
+        totalAccuracy = totalAccuracy + gearBoost
+    end
+
+    -- Check to see if the player has any elemental affinity gear like an elemental staff.
+    local elementalAccBonus = caster:getMod(xi.combat.element.elementalMagicAcc[spellElement]) + caster:getMod(xi.combat.element.strongAffinityAcc[spellElement]) * 10
```

Comment:
Since you are calling `applyResistanceEffect(caster, target, spell, resistParams)` at a later step, this is unneeded.